pkgs/test/nixpkgs-check-by-name/.envrc

```diff
@@ -1,7 +1,7 @@
-if has nix_direnv_watch_file; then
-  nix_direnv_watch_file default.nix shell.nix
-else
+if has watch_file; then
   watch_file default.nix shell.nix
+else
+  nix_direnv_watch_file default.nix shell.nix
 fi
 
```

Comment:
I believe we can actually remove the entire thing new!

```suggestion
```

This was originally necessary due to combination of https://github.com/nix-community/nix-direnv/issues/409 and https://github.com/nix-community/nix-direnv/issues/408, but these are fixed now (which also indirectly caused the deprecation of `nix_direnv_watch_file`!)

I don't have these direnv fixes installed locally yet so I can't check that it fully works myself (and indeed will cause problems for me until I update direnv, but that's fine). Can you verify that this works by patching the `default.nix` file like this:

```diff
diff --git a/pkgs/test/nixpkgs-check-by-name/default.nix b/pkgs/test/nixpkgs-check-by-name/default.nix
index 8836da1f403f..77a0b6f8ae9e 100644
--- a/pkgs/test/nixpkgs-check-by-name/default.nix
+++ b/pkgs/test/nixpkgs-check-by-name/default.nix
@@ -60,6 +60,7 @@ let
           --set NIX_CHECK_BY_NAME_EXPR_PATH "$NIX_CHECK_BY_NAME_EXPR_PATH"
       '';
       passthru.shell = mkShell {
+        env.SOME_ENV = "test";
         env.NIX_CHECK_BY_NAME_EXPR_PATH = toString runtimeExprPath;
         env.NIX_PATH = "test-nixpkgs=${toString testNixpkgsPath}:test-nixpkgs/lib=${toString nixpkgsLibPath}";
         inputsFrom = [ package ];
```

Then pressing e.g. Enter in a terminal, making sure that the environments gets reloaded, then run

```
$ echo $SOME_ENV
```

Which should output `test` :)

Reply:
I checked and it is working

```
pkgs/test/nixpkgs-check-by-name on  master [!] is 📦 v0.1.0 via 🦀 v1.76.0 via ❄️  impure (nix-shell-env)
❯ echo $SOME_ENV
test
```